sawtooth-daml-tp/src/main/java/com/blockchaintp/sawtooth/daml/processor/impl/DamlTransactionHandler.java

```diff
@@ -204,10 +207,10 @@ private Configuration getDefaultConfiguration() throws InternalError, InvalidTra
     return Arrays.asList(new String[] {this.namespace });
   }
 
-  private Timestamp getRecordTime(final LedgerState ledgerState) throws InternalError {
+  private Timestamp getRecordTime(final LedgerState ledgerState) throws InternalError, IllegalArgumentException {
```

Comment:
IllegalArgumentException is an unchecked exception, don't declare it as thrown unless it happens in some obscure circumstance.  In this particular case, it should probably be wrapped in an InternalError and thrown, since it would indicate something very wrong happened in the system.

Reply:
removed
the assertFromLong may throw a scala IllegalArgumentException - highly unlikely, but it is a potentiality

Reply:
i went back to the old syntax and although the IDE complains it seems to compile